core/src/main/scala/io/finch/EndpointResult.scala

```diff
@@ -93,11 +94,15 @@ sealed abstract class EndpointResult[+A] {
 
 object EndpointResult {
 
-  case object Skipped extends EndpointResult[Nothing] {
+  final case class Matched[A](rem: Input, out: Rerunnable[Output[A]]) extends EndpointResult[A] {
+    def isMatched: Boolean = true
+  }
+
+  case object NotMatched extends EndpointResult[Nothing] {
```

Comment:
I want to look into this to see if it could be modeled better so we can avoid `asInstanceOf` casts in `Endpoint`.